### PR TITLE
misc: fix for On Demand Chunk page-in span

### PR DESCRIPTION
**Pull request by sherali42**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?

Fix to accurately capture the latency for On Demand Paging of RawChunks. Span is created only if there are partitions that needs to be paged-in from datastore.

### CODE REVIEW

**Comment by broneill** [comment]
The change looks scarier than it actually is, but I don't understand it. Can you summarize how the change works?

**Comment by sherali42** [comment]
> The change looks scarier than it actually is, but I don't understand it. Can you summarize how the change works?

- Span creation is moved inside the conditional block, so that it gets created only if there are chunks that needs to be paged in. 
- Span is closed on completion of the task as it is an observable. `.doOnComplete(() => span.finish())`
- Also moved `Observable.fromIterable(inMemoryPartitions) ++` outside, as we are doing this in both `if` and `else`

Earlier code was measuring the latency regardless of if we are actually ODPing.

File: core/src/main/scala/filodb.core/memstore/OnDemandPagingShard.scala [hunk 1]
```diff
@@ -73,42 +73,48 @@ TimeSeriesShard(dataset, storeConfig, shardNum, rawStore, metastore, evictionPol
...
-    } else {
-      Observable.fromIterable(inMemoryPartitions) ++
-      Observable.fromTask(odpPartTask(indexIt, partKeyBytesToPage, methods, chunkMethod)).flatMap { odpParts =>
-        shardStats.partitionsQueried.increment(partKeyBytesToPage.length)
-        Observable.fromIterable(partKeyBytesToPage.zip(methods))
-          .mapAsync(storeConfig.demandPagingParallelism) { case (partBytes, method) =>
-            rawStore.readRawPartitions(dataset, allDataCols, SinglePartitionScan(partBytes, shardNum), method)
+    Observable.fromIterable(inMemoryPartitions) ++ {
+      if (storeConfig.multiPartitionODP) {
+        Observable.fromTask(odpPartTask(indexIt, partKeyBytesToPage, methods, chunkMethod)).flatMap { odpParts =>
```
Comment (vishramachandran) [inline core/src/main/scala/filodb.core/memstore/OnDemandPagingShard.scala]: We also need a span in createODPPartitionsTask method to measure how much time it took to create the partitions in memory.
